_posts/2024-01-11-gsoc2023_xhgchen.md

```diff
@@ -0,0 +1,157 @@
+
+- Implement a class to calculate self-diffusivity via the Green-Kubo method
+- Write a utility function to calculate shear viscosity via the Einstein method
+- Create a utility function to calculate shear viscosity via the Green-Kubo method
+- Test and document the new features to ensure they are reliable and easy to use
+
+After some discussion that began with a suggestion from @jaclark5 to try the Helfand method for viscosity to better integrate the feature wtih MDAnalysis, we shifted our priorities away from the utility functions to focus on a new Einstein-Helfand viscosity class. That extra research and trial and error were well worth the result - better software with a more simple and robust API.
+
+## My Summer in Python and NumPy Arrays
+
+My summer culminated in the creation of [Transport Analysis](https://github.com/MDAnalysis/transport-analysis), a standalone [MDAKit](https://mdakits.mdanalysis.org/transport-analysis.html) Python package to compute and analyze transport properties powered by MDAnalysis.
+
+Transport Analysis GitHub: <https://github.com/MDAnalysis/transport-analysis>
+
+### Building a FAIR-compliant Python Package from the MDAKit Cookiecutter Template
+
+[Generating my package](https://github.com/MDAnalysis/transport-analysis/commit/1814818e6922cd7351ff1aa265e2f3f140b4ba0a) from the [MDAKit cookiecutter template](https://mdakits.mdanalysis.org/makingakit.html#building-from-the-cookiecutter) streamlined package creation, ensuring that Transport Analysis was set up through a reliable, stringently reviewed process. The included CI/CD pipeline needed some adjusting on GitHub, but it facilitated writing maintainable software from the very beginning. [Pull request (PR) #3](https://github.com/MDAnalysis/transport-analysis/pull/3) added the new code formatter *Black* to the CI, while [PR #22](https://github.com/MDAnalysis/transport-analysis/pull/22) flattened the package's structure to simplify its organization of modules.
+
+### Velocity Autocorrelation Functions (VACFs) and Green-Kubo Self-Diffusivity
+
+[PR #1](https://github.com/MDAnalysis/transport-analysis/pull/1) built two implementations of the VACF calculation: a simple "windowed" algorithm and a fast FFT algorithm leveraging `tidynamics`. All the computations involved are vectorized, but the FFT algorithm speeds the calculations up even further. I designed a simple test system from a unit velocity trajectory and began with some very basic calculations on pen and paper, then gradually wrote more and more tests until my code was tested with the same rigour as the [MDAnalysis MSD module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/msd.html) on top of reaching 100% code coverage.

```

Comment:
```suggestion
[PR #1](https://github.com/MDAnalysis/transport-analysis/pull/1) built two implementations of the VACF calculation: a simple "windowed" algorithm and a fast FFT-based algorithm leveraging `tidynamics`. All the computations involved are vectorized, but the FFT-based algorithm speeds the calculations up even further. I designed a simple test system from a unit velocity trajectory and began with some very basic calculations on pen and paper, then gradually wrote more and more tests until my code was tested with the same rigour as the [MDAnalysis MSD module](https://docs.mdanalysis.org/stable/documentation_pages/analysis/msd.html) on top of reaching 100% code coverage.

```
?

Reply:
@marinegor - Is it accurate to say 'FFT-_based_' here?

For the sake of efficiency, I've also accepted @RMeli's other suggestions as they were minor. However, this is ultimately your blog post, @marinegor, so please have a look and make sure you are okay with the changes.

Reply:
yes, it's ok! I already had a look and the text is completely fine -- sorry I wasn't more explicit about it.

Reply:
Thanks, @marinegor! And sorry for pinging you on this one - I just realized the FFT-based clarification should have been directed to @xhgchen

Reply:
Yes, FFT-based is good too! Appreciate all the feedback and will have to update the post on my website since it was originally written before the package came out on PyPI. Sorry for the delay - I am still sick and have barely had a chance to breathe since flying back and getting straight to work.

Reply:
Sorry to hear that you are still not feeling well - get better soon!